script.trakt/kodiUtilities.py

```diff
@@ -0,0 +1,366 @@
+    logger.debug("getEpisodeDetailsFromKodi(): %s" % str(result))
+
+    if not result:
+        logger.debug("getEpisodeDetailsFromKodi(): Result from Kodi was empty.")
+        return None
+
+    show_data = getShowDetailsFromKodi(result['episodedetails']['tvshowid'], ['year', 'imdbnumber'])
+
+    if not show_data:
+        logger.debug("getEpisodeDetailsFromKodi(): Result from getShowDetailsFromKodi() was empty.")
+        return None
+
+    result['episodedetails']['imdbnumber'] = show_data['imdbnumber']
+    result['episodedetails']['year'] = show_data['year']
+
+    try:
+        return result['episodedetails']
+    except KeyError:
+        logger.debug("getEpisodeDetailsFromKodi(): KeyError: result['episodedetails']")
+        return None
+
+# get a single movie from kodi given the id
+def getMovieDetailsFromKodi(libraryId, fields):
+    result = kodiJsonRequest({'jsonrpc': '2.0', 'method': 'VideoLibrary.GetMovieDetails', 'params': {'movieid': libraryId, 'properties': fields}, 'id': 1})
+    logger.debug("getMovieDetailsFromKodi(): %s" % str(result))
+
+    if not result:
+        logger.debug("getMovieDetailsFromKodi(): Result from Kodi was empty.")
+        return None
+
+    try:
+        return result['moviedetails']
+    except KeyError:
+        logger.debug("getMovieDetailsFromKodi(): KeyError: result['moviedetails']")
+        return None
+
+def checkAndConfigureProxy():
+    proxyActive = kodiJsonRequest({'jsonrpc': '2.0', "method":"Settings.GetSettingValue", "params":{"setting":"network.usehttpproxy"}, 'id': 1})['value']
+    proxyType = kodiJsonRequest({'jsonrpc': '2.0', "method":"Settings.GetSettingValue", "params":{"setting":"network.httpproxytype"}, 'id': 1})['value']
+
+    if proxyActive and proxyType == 0: # PROXY_HTTP
+        proxyURL = kodiJsonRequest({'jsonrpc': '2.0', "method":"Settings.GetSettingValue", "params":{"setting":"network.httpproxyserver"}, 'id': 1})['value']
+        proxyPort = unicode(kodiJsonRequest({'jsonrpc': '2.0', "method":"Settings.GetSettingValue", "params":{"setting":"network.httpproxyport"}, 'id': 1})['value'])
+        proxyUsername = kodiJsonRequest({'jsonrpc': '2.0', "method":"Settings.GetSettingValue", "params":{"setting":"network.httpproxyusername"}, 'id': 1})['value']
+        proxyPassword = kodiJsonRequest({'jsonrpc': '2.0', "method":"Settings.GetSettingValue", "params":{"setting":"network.httpproxypassword"}, 'id': 1})['value']
+
+        if proxyUsername and proxyPassword and proxyURL and proxyPort:
+            regexUrl = re.compile(REGEX_URL)
+            matchURL = regexUrl.search(proxyURL)
+            if matchURL:
+                return matchURL.group(1) + proxyUsername + ':' + proxyPassword + '@' + matchURL.group(2) + ':' + proxyPort
+        elif proxyURL and proxyPort:
+            return proxyURL + ':' + proxyPort
+    
+    return None
+
+def getMediaType():
```

Comment:
this one here probably wont work anymore for all cases, you probably have to switch to checkin the mediatype of the listitem instead of the container content type.